Netkan/Processors/QueueHandler.cs

```diff
@@ -0,0 +1,206 @@
+{
+    public class QueueHandler
+    {
+        public QueueHandler(string cacheDir, bool overwriteCache, string githubToken, bool prerelease)
+        {
+            log.Debug("Initializing SQS queue handler");
+            // TODO: Shouldn't hard code the region
+            client   = new AmazonSQSClient(RegionEndpoint.USWest2);
+            inflator = new Inflator(cacheDir, overwriteCache, githubToken, prerelease);
+
+            inputQueueURL  = getQueueUrl(inputQueueName);
+            outputQueueURL = getQueueUrl(outputQueueName);
+            log.DebugFormat("Queue URLs: {0}, {1}", inputQueueURL, outputQueueURL);
+        }
+
+        public void Process()
+        {
+            while (true)
+            {
+                int releases;
+                Metadata netkan = getNetkan(out releases);
+                if (netkan == null)
+                {
+                    continue;
+                }
+                log.InfoFormat("Inflating {0}", netkan.Identifier);
+                IEnumerable<Metadata> ckans = null;
+                try
+                {
+                    ckans = inflator.Inflate($"{netkan.Identifier}.netkan", netkan, releases);
+                }
+                catch (Exception e)
+                {
+                    e = e.GetBaseException() ?? e;
+                    log.InfoFormat("Inflation failed, sending error: {0}", e.Message);
+                    sendCkan(null, netkan, false, e.Message);
+                }
+                if (ckans != null)
+                {
+                    foreach (Metadata ckan in ckans)
+                    {
+                        log.InfoFormat("Sending {0}-{1}", ckan.Identifier, ckan.Version);
+                        sendCkan(ckan, netkan, true);
+                    }
+                }
+            }
+        }
+
+        private string getQueueUrl(string name)
+        {
+            log.DebugFormat("Looking up URL for queue {0}", name);
+            return client.GetQueueUrl(new GetQueueUrlRequest() { QueueName = name }).QueueUrl;
+        }
+
+        private Metadata getNetkan(out int releases)
+        {
+            log.Debug("Retrieving metadata from queue");
+            // TODO: Get this from an attribute
+            releases = 1;
+            var msg = getFromQueue(inputQueueURL);
+            if (msg != null)
+            {
+                log.DebugFormat("Metadata returned: {0}", msg.Body);
+                return new Metadata(JObject.Parse(msg.Body));
+            }
+            log.Debug("No metadata in queue");
+            return null;
+        }
+
+        private Message getFromQueue(string url)
+        {
+            log.DebugFormat("Looking for message from {0}", url);
+            var resp = client.ReceiveMessage(new ReceiveMessageRequest()
+            {
+                QueueUrl            = url,
+                MaxNumberOfMessages = 1,
+                VisibilityTimeout   = (int)TimeSpan.FromMinutes(15).TotalSeconds,
```

Comment:
This is really good. If we run multiple inflators we can use this to avoid duplication of inflation.

Reply:
Yup (and apparently [it's required if you want to be able to delete messages as well](https://github.com/aws/aws-sdk-js/issues/1279#issuecomment-268830676)).

What about `MaxNumberOfMessages`? Should we try to handle more?

Reply:
> Yup (and apparently [it's required if you want to be able to delete messages as well](https://github.com/aws/aws-sdk-js/issues/1279#issuecomment-268830676)).
Ohhh, interesting. Good find!

> What about `MaxNumberOfMessages`? Should we try to handle more?
It would be more efficient message wise to handle them in batches of 10. We would need to consider visibility timeouts and what not. If anything maybe allow for multiple, but set max messages to 1. Then when we've shaken out the logistics our basic layout is for multiple messages.

Reply:
Refactored to get 10 at once. It seems to work.